todo_app/app.py

```diff
@@ -1,57 +1,159 @@
+        result = list_cards.get_List(id = item['id']) 
+        if result.status_code == 200:
+            for card in result.json():
+                if not (card['due'] is None):
+                    formated_date = (card['due']).split("T")[0]
+                else:
+                    formated_date = card['due']
+                cards.append ({
+                    'id' : card['id'],
+                    'title': card['name'],
+                    'status' : item['status'],
+                    'due' : formated_date
+                })
+        else:
+            return render_template("404.html",error="failed to get Trello cards!")    
+    return render_template('index.html', tasks=cards)
 
-#new task
+# new task
 @app.route('/new', methods=['GET'])
 def getnew_post():
    return render_template('new_task.html')
 
-#edit task
-@app.route('/edit/<int:id>', methods=['GET'])
+@app.route('/', methods=['POST'])
+def post_index():    
+    card = TrelloCard()
+    result = card.create_Card(
+        name = request.form['title'],
+        due = request.form['duedate'],
+        desc = request.form['descarea'],
+        id = get_listId(status='Not Started')
+    )
+    
+    if (result.status_code == 200):
+        return redirect('/')
+    else:
+        return render_template("404.html",error="failed to create Trello card!")
+
+# edit task
+@app.route('/edit/<id>', methods=['GET'])
 def get_edit(id):
-    task = get_item(id)
-    return render_template('edit.html', task=task)
+    card = TrelloCard()
+    result = card.get_Card(id=id)
+    if (result.status_code == 200):
+        card_info = Card(
+           json = result.json(),
+           statusName = get_statusName(id=result.json()['idList'])
+        )
+        return render_template('edit.html', task=card_info)
+    else:
+        return render_template("404.html", error="failed to obtain Trello card info!")
 
-@app.route('/edit/<int:id>', methods=['POST'])
+@app.route('/edit/<id>', methods=['POST'])
 def post_edit(id):
-    task = get_item(id)
-    task['title'] = request.form['title']
-    task['status'] = request.form['status']
-    save_item(task)
-    return redirect('/')
-
-#delete task
-@app.route('/delete/<int:id>')
+    card = TrelloCard()
+    result = card.update_Card(
+        id = id,
+        listId = get_listId(status=request.form['status']),
+        name = request.form['title'],
+        desc = request.form['descarea'],
+        due = request.form['duedate']
+    )
+    if result.status_code == 200:
+        return redirect('/')
+    else:
+        return render_template("404.html", error="failed to update Trello card!")
+
+# delete task
+@app.route('/delete/<id>')
 def delete(id):
-    delete_item(id)
-    return redirect('/')
+    card = TrelloCard()
+    result = card.delete_Card(id=id)
+    if result.status_code == 200:
+        return redirect('/')
+    else:
+        return render_template("404.html",error="failed to delete Trello card!")
```

Comment:
It would be nice to update the name of your error page since you're using it for wider issues than just a 404 now

Reply:
noted, thank you